python/lsst/fgcmcal/fgcmCalibrateTract.py

```diff
@@ -20,64 +20,21 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
-"""Run fgcmcal on a single tract.
-
+"""Class for running fgcmcal on a single tract using src tables.
 """
 
-import sys
-import traceback
-
-import numpy as np
-
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
 from lsst.jointcal.dataIds import PerTractCcdDataIdContainer
 
-from .fgcmBuildStars import FgcmBuildStarsTask
-from .fgcmFitCycle import FgcmFitCycleConfig
-from .fgcmOutputProducts import FgcmOutputProductsTask
-from .utilities import makeConfigDict, translateFgcmLut, translateVisitCatalog
-from .utilities import computeCcdOffsets, computeApertureRadiusFromSchema, extractReferenceMags
-from .utilities import makeZptSchema, makeZptCat
-from .utilities import makeAtmSchema, makeAtmCat
-from .utilities import makeStdSchema, makeStdCat
-
-import fgcm
-
-__all__ = ['FgcmCalibrateTractConfig', 'FgcmCalibrateTractTask', 'FgcmCalibrateTractRunner']
+from .fgcmCalibrateTractBase import (FgcmCalibrateTractConfigBase, FgcmCalibrateTractRunner,
+                                     FgcmCalibrateTractBaseTask)
 
+__all__ = ['FgcmCalibrateTractConfig', 'FgcmCalibrateTractTask']
 
-class FgcmCalibrateTractConfig(pexConfig.Config):
-    """Config for FgcmCalibrateTract"""
-
-    fgcmBuildStars = pexConfig.ConfigurableField(
-        target=FgcmBuildStarsTask,
-        doc="Task to load and match stars for fgcm",
-    )
-    fgcmFitCycle = pexConfig.ConfigField(
-        dtype=FgcmFitCycleConfig,
-        doc="Config to run a single fgcm fit cycle",
-    )
-    fgcmOutputProducts = pexConfig.ConfigurableField(
-        target=FgcmOutputProductsTask,
-        doc="Task to output fgcm products",
-    )
-    convergenceTolerance = pexConfig.Field(
-        doc="Tolerance on repeatability convergence (per band)",
-        dtype=float,
-        default=0.005,
-    )
-    maxFitCycles = pexConfig.Field(
-        doc="Maximum number of fit cycles",
-        dtype=int,
-        default=5,
-    )
-    doDebuggingPlots = pexConfig.Field(
-        doc="Make plots for debugging purposes?",
-        dtype=bool,
-        default=False,
-    )
 
+class FgcmCalibrateTractConfig(FgcmCalibrateTractConfigBase):
+    """Config for FgcmCalibrateTract task"""
```

Comment:
Because the two classes (`Table` and `not Table`) have different configs, I don't know how to navigate the default setting other than this slight duplication.  I think when the `not Table` code is deprecated in the future things will be simplified.

Reply:
Maybe add comments to each of the changed default lines, so it's more clear why they are different from the parent class?